### PR TITLE
Rework creating jobs

### PR DESCRIPTION
**Pull request by warewarewapain**

### CODE REVIEW

File: hammett/core/exceptions.py [hunk 1]
```diff
@@ -11,6 +15,10 @@ class FailedToGetDataAttributeOfQuery(Exception):
     """Raised when the attempt to get a data attribute of a query fails."""
 
 
+class JobKwargsHaveNotBeenProvided(Exception):
```
Comment (eugulixes) [inline hammett/core/exceptions.py]: Keep it simple: `JobKwargsNotProvided`

File: hammett/core/exceptions.py [hunk 2]
```diff
@@ -11,6 +15,10 @@ class FailedToGetDataAttributeOfQuery(Exception):
...
+    """Raised when the job_kwargs for the job has not been provided."""
```
Comment (eugulixes) [inline hammett/core/exceptions.py]:
```
Raised when `job_kwargs` for a job has not been provided.
```

File: hammett/core/exceptions.py [hunk 3]
```diff
@@ -1,6 +1,10 @@
 """The module contains the global Hammett exception and warning classes."""
 
 
+class CallbackForJobHasNotBeenProvided(Exception):
+    """Raised when the callback function for the job has not been provided."""
```
Comment (eugulixes) [inline hammett/core/exceptions.py]:
```
Raised when a callback function for a job has not been provided.
```

File: hammett/core/application.py [hunk 4]
```diff
@@ -135,22 +140,30 @@ def _register_error_handlers(
...
-                if job_queue:
-                    job_queue.run_repeating(
-                        handler,
-                        first=first_request,
-                        interval=interval_request,
+                job_kwargs = job_queue_handler.get('job_kwargs')
+                if not job_kwargs:
+                    msg = (
+                        f'You must provide the keyword arguments (job_kwargs) for the '
+                        f'configuration of the job with index {i}'
```
Comment (eugulixes) [inline hammett/core/application.py]: What about this short version of the message?

> You must provide job_kwargs for the job under the index {i}

File: hammett/core/application.py [hunk 5]
```diff
@@ -135,22 +140,30 @@ def _register_error_handlers(
...
-            for job_queue_handler in job_queue_handlers:
-                handler = job_queue_handler['handler']
-                first_request = job_queue_handler['first_request']
-                interval_request = job_queue_handler['interval_request']
+            for i, job_queue_handler in enumerate(job_queue_handlers):
+                callback = job_queue_handler.get('callback')
+                if callback is None:
+                    msg = (
+                        f'You must provide the callback function that will be executed by the job '
+                        f'in configuration with index {i}'
```
Comment (eugulixes) [inline hammett/core/application.py]: > You must provide a callback function that will be executed by the job under the index {i}

**Review by Themanwhosmellslikesugar — APPROVED** [review]

**Pull request merged**